### PR TITLE
Add changelog appendix

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

**Comment by pietroalbini** [comment]
bors try

File: src/changelog.rst [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  * While updating the FLS to account for this feature, we realized that the
+    way the FLS described crate types was incorrect. We rectified this:
+
+    * New section: :ref:`fls_8JB3SJqamdpU`
+    * New glossary entry: :t:`crate type`
+    * New paragraphs: :p:`fls_unxalgMqIr3v`, :p:`fls_e7jGvXvTsFpC`, :p:`fls_kQiJPwb2Hjcc`, :p:`fls_OyFwBtDGVimT`
+    * Updated glossary entries: :t:`binary crate`, :t:`proc-macro crate`
+    * Updated paragraphs: :p:`fls_9ub6ks8qrang`, :p:`fls_Mf62VqAhoZ3c`
+    * Moved paragraph: :p:`fls_sbGnkm8Ephiu`
+    * Removed paragraph about library crates
```
Comment (pietroalbini) [inline src/changelog.rst:77]: Re-reading the change to reword this section, it feels wrong that we removed the description of library crates in the FLS. Can you send a follow-up PR to add back that paragraph and glossary entry?

File: src/changelog.rst [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+* `Check return types of function types for well-formedness <https://github.com/rust-lang/rust/pull/115538>`_
+
+  * No change: the exact trait resolution implementation is not part of the FLS
+
+* `Rework \`impl Trait\` lifetime inference <https://github.com/rust-lang/rust/pull/116891/>`_
+
+  * No change: capturing of lifestime within ``impl Trait`` types is not described in the FLS
```
Comment (pietroalbini) [inline src/changelog.rst:85]: This feels like a gap in the FLS we should fill (and describe).

File: src/changelog.rst [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+* `Remove unnecessary type inference when using associated types inside of higher ranked \`where\`-bounds <https://github.com/rust-lang/rust/pull/119849>`_
+
+  * No change: the FLS does not specify type inference to such a degree
+
+* `Weaken eager detection of cyclic types during type inference <https://github.com/rust-lang/rust/pull/119989>`_
+
+  * No change: the FLS does not specify type inference to such a degree
+
+* `\`trait Trait: Auto {}\`: allow upcasting from \`dyn Trait\` to \`dyn Trait + Auto\` <https://github.com/rust-lang/rust/pull/119338>`_
```
Comment (pietroalbini) [inline src/changelog.rst:146]: The description of what changed here is missing.

**Review by pietroalbini — CHANGES_REQUESTED** [review]
Pushed some commits to reword and reformat the changelog entries a bit. I also configured the lint to prevent having paragraph IDs in this page, which just add noise for no benefit.

Left comments on things that I feel we need to expand on.

**Comment by pietroalbini** [comment]
bors try

**Comment by pietroalbini** [comment]
bors try

**Review by pietroalbini — APPROVED** [review]
Marking Lukas's changes as approved, waiting for Lukas to approve my changes before we merge this.

I think the best thing is to merge this PR right now, and tackle the three points I mentioned in https://github.com/ferrocene/specification/pull/511#pullrequestreview-2158659401 in followup PRs that both fix the points and update the changelog.

**Review by Veykril — COMMENTED** [review]
👍 (cant approve my own PR)

**Comment by pietroalbini** [comment]
bors merge

**Pull request merged**